### PR TITLE
ns for fx: remove torch.ops.quantized.cat

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #76993 ns for fx: remove duplicated BNReLU mappings
* #76992 ns for fx: remove quantized ReLU6 from mapping
* #76991 ns for fx: remove GroupNorm from mapping
* #76990 ns for fx: clean up linear in relationship mapping
* #76989 ns for fx: make torch.ops.quantized.dropout mapping dynamic
* **#76988 ns for fx: remove torch.ops.quantized.cat**
* #76980 ns for fx: clean up convtranspose mappings
* #76978 ns for fx to backend_config_dict [2/x]: native lowering mappings
* #76958 ns for fx to backend_config_dict [1/x]: fused and qat modules

Summary:

FX graph mode quantization no longer uses `torch.ops.quantized.cat`,
instead `torch.cat` can use quantized inputs.

This PR removes the outdated mapping from NS for FX.

Test plan:

```
python test/test_quantization.py -k FXNumericSuite
```

Differential Revision: [D36215853](https://our.internmc.facebook.com/intern/diff/D36215853)

### CODE REVIEW

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215853).

**Review by jerryzh168 — APPROVED** [review]

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215853).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215853).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215853).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215853).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215853).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215853).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D36215853).
